### PR TITLE
Clarify double-dot path segment parsing

### LINKED ISSUE #295: "append the empty string to url’s path." is confusing
https://url.spec.whatwg.org/commit-snapshots/488c459d9e4245a3f6bf087e7dcd2c7e91487ac5/#url-parsing

> append the empty string to url’s path.

It would be nice if the text explained why we are doing this apparently pointless step.

### PR DESCRIPTION
**Pull request by annevk**
Fixes #295.

@GPHemsley @Hixie, what do you think?

***
<a href="https://whatpr.org/url/482.html" title="Last updated on Apr 28, 2020, 6:17 AM UTC (3f408d6)">Preview</a> | <a href="https://whatpr.org/url/482/980ba89...3f408d6.html" title="Last updated on Apr 28, 2020, 6:17 AM UTC (3f408d6)">Diff</a>

### CODE REVIEW

File: url.bs [hunk 1]
```diff
@@ -2099,10 +2099,19 @@ string <var>input</var>, optionally with a <a>base URL</a> <var>base</var>, opti
...
+
+        <ol>
+         <li><p><a>Shorten</a> <var>url</var>'s <a for=url>path</a>.
+
+         <li>
+          <p>If neither <a>c</a> is U+002F (/), nor <var>url</var> <a>is special</a> and <a>c</a> is
+          U+005C (\), <a for=list>append</a> the empty string to <var>url</var>'s
+          <a for=url>path</a>.
+
+          <p class="note no-backref">Usually <a>c</a> is U+002F (/) at this point.
```
Comment (domenic) [inline url.bs:2113]: This doesn't really help much IMO. Addressing #482 would involve explaining the meaning of empty path segments, either here or somewhere else that this step could link to.

Comment (annevk) [reply url.bs:2113]: How are they different from other path segments?

Comment (domenic) [reply url.bs:2113]: Empty is often a special case, at least in human's minds. I agree for computers they are not different.

Comment (annevk) [reply url.bs:2113]: Okay, would something like "This means that for input `/foo/..` the result is `/foo/` and not `/foo`." work?

Comment (domenic) [reply url.bs:2113]: Yeah that makes a lot of sense to me.

**Pull request merged**